vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +812,42 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public void writeInferredFacts(OutputStream stream) throws IOException {
+		HashSet<Predicate> toBeQueriedHeadPredicates = new HashSet<Predicate>();
+		for (Rule rule : this.knowledgeBase.getRules()) {
+			for (Literal literal : rule.getHead()) {
+				toBeQueriedHeadPredicates.add(literal.getPredicate());
+			}
+		}
+		for (Fact fact : this.knowledgeBase.getFacts()) {
+			if (!toBeQueriedHeadPredicates.contains(fact.getPredicate()))
+				stream.write((fact.toString() + "\n").getBytes());
+		}
+		for (Predicate predicate : toBeQueriedHeadPredicates) {
+			ArrayList<Term> tobeGroundedVariables = new ArrayList<Term>();
+			for (int i = 0; i < predicate.getArity(); i++) {
+				tobeGroundedVariables.add(Expressions.makeUniversalVariable("X" + i));
+			}
+			final QueryResultIterator answers = this

```

Comment:
use try-with-resources for the `QueryResultIterator `
(for example, look for how is use it in the vlog4j-examples module)